pkg/apis/core/helper/helpers.go

```diff
@@ -133,6 +134,41 @@ func TaintsHave(taints []core.SeedTaint, key string) bool {
 	return false
 }
 
+// TaintsAreTolerated returns true when all the given taints are tolerated by the given tolerations. It ignores the
+// deprecated taints that were migrated into the new `settings` field in the Seed specification.
+func TaintsAreTolerated(taints []core.SeedTaint, tolerations []core.Toleration) bool {
+	var relevantTaints []core.SeedTaint
+	for _, taint := range taints {
+		if taint.Key == core.DeprecatedSeedTaintDisableDNS || taint.Key == core.DeprecatedSeedTaintInvisible || taint.Key == core.DeprecatedSeedTaintDisableCapacityReservation {
+			continue
+		}
+		relevantTaints = append(relevantTaints, taint)
+	}
+
+	if len(relevantTaints) == 0 {
+		return true
+	}
+	if len(tolerations) == 0 {
+		return false
+	}
+
+	taintKeyValues := sets.NewString()
+	for _, taint := range relevantTaints {
+		taintKeyValues.Insert(utils.IDForKeyWithOptionalValue(taint.Key, taint.Value))
+	}
+
+	for _, toleration := range tolerations {
+		if taintKeyValues.Has(toleration.Key) {
+			continue
+		}
+		if !taintKeyValues.Has(utils.IDForKeyWithOptionalValue(toleration.Key, toleration.Value)) {
```

Comment:
This means that all taints must contain all tolerations but it should be the other way around, shouldn't it? At the moment, the seeds isn't accepted if the shoots has more tolerations than the seed has taints.

Reply:
Very nice catch, thanks for finding it, indeed. I've inverted the logic as you outlined and added another test case for it.